dlt/sources/rest_api/__init__.py

```diff
@@ -263,12 +263,17 @@ def create_resources(
             incremental_cursor_transform,
         ) = setup_incremental_object(request_params, endpoint_config.get("incremental"))
 
+        merged_client_config: ClientConfig = {
```

Comment:
configs get merged here, with the keys defined in the endpoint taking preference over the global one.